services/dune.tsx

```diff
@@ -0,0 +1,73 @@
+import {strict as assert} from 'node:assert'
+
+const DUNE_API_KEY = process.env.DUNE_API_KEY
+assert(DUNE_API_KEY, "DUNE_API_KEY environment var is required")
+
+// TODO: getFromDune will be moved in a future PR to the SDK
+interface MetadataQuery {
+  executed_at: string,
+  job_id: string,
+  query_duration_millis: number,
+  query_version: number,
+  result_bytes: number
+  result_rows: number
+}
+
+interface GetFromDuneResult<T> {
+  metadata: MetadataQuery
+  column_names: string[]
+  rows: T[]
+}
+
+export async function getFromDune<T>(queryId: number): Promise<GetFromDuneResult<T>> {
+  const response = await fetch(`https://api.dune.com/api/v0/query/${queryId}/results`, {
+    headers: {
+      accept: "application/json",
+      "X-DUNE-API-KEY": DUNE_API_KEY
+    }
+  })
+
+  return await response.json()
+}
+
+// ------ End of TODO
+
+const TOTAL_TRADES_COUNT_QUERY_ID = 1034337
+const TOTAL_SURPLUS_COUNT_QUERY_ID = 270604
+
+
+interface TotalCount {
+  totalCount: number
+  lastModified: Date
+}
+
+export async function _getTotalCount(queryId: number): Promise<TotalCount> {
+  const queryResut = await getFromDune<{ count: number }>(queryId)
+  console.log('queryResut', queryId, queryResut)
+
+  // Expect one row
+  assert(
+    queryResut.rows.length === 1, 
+    `Total Count Dune query (${queryId}) must return just one row. Returned ${queryResut.rows.length}`
+  )
```

Comment:
What happens if this fails? How will we see the error?
I assume the server will return a 503 or something like that?

Reply:
I don't think so, if it's a temporal error, i would think it uses the cache. Anyways, this deserves a test in another PR